### ISSUE TITLE
Dark mode does not activate with GPS disabled

### ISSUE THREAD

**Issue by BikeRoad** [osmandapp/OsmAnd#15230]
During the night the "map mode: sunrise/sunset" function does not automatically switch to dark mode when gps is off. If the gps is activated then it works correctly.

In addition, this problem occurs with any of the osmand map styles.

I have tested this on two different phones with different versions of android.
Version Osmand 4.2.7

**Comment by sonora** [comment]
How do you expect to determine when sunrise/sunset while your location is not determined?! 😉

**Comment by BikeRoad** [comment]
As seen in the image, there is a pre-configured time and I thought that the switch to dark mode is based on the system time (android) and not on the location.
![Map mode](https://user-images.githubusercontent.com/19896035/188328407-77b41eb7-1492-4372-9eef-23babf23b712.jpg)

**Comment by sonora** [comment]
Good point, thx! Those must be cached times while the actual switch is likely based on current location, and being suppressed if none is known.

I will leave it to product management to judge if this is considered a bug.
